roofit/roofitcore/src/RooAbsArg.cxx

```diff
@@ -1623,12 +1623,29 @@ void RooAbsArg::optimizeCacheMode(const RooArgSet& observables, RooArgSet& optim
 
 
   // Terminate call if this node was already processed (tree structure may be cyclical)
-  if (processedNodes.findArg(this)) {
-    return ;
-  } else {
-    processedNodes.Add(this) ;
+  // LM : RooLinkedList::findArg looks by name and not but by object pointer,
```

Comment:
Hi @lmoneta, are all these commented out lines still necessary? They look like they have been relevant in the debugging process, but maybe it's better to have just one line of comment about why `FindObject` is important here?

Reply:
The original idea was to print a warning message for nodes with  duplicated names, but them I noticed this happens quite frequently. I think is not a too bad idea to keep this commented out, in case we want to remember to print the message if needed again.

Reply:
Okay, thanks for the exaplanation

Reply:
I have now updated the code and keep the message but on the INFO stream that by default is not printed. 
I think  like this is better.

Reply:
Thanks, I also think that if we have the optional logging we should make use of it and prefer this to commented out `cout`s.